airflow/contrib/hooks/azure_batchai_hook.py

```diff
@@ -0,0 +1,99 @@
+
+    def __init__(self, azure_batchai_conn_id='azure_batchai_default'):
+        self.conn_id = azure_batchai_conn_id
+        self.connection = self.get_conn()
+
+    def get_conn(self):
+        conn = self.get_connection(self.conn_id)
+        key_path = conn.extra_dejson.get('key_path', False)
+        if key_path:
+            if key_path.endswith('.json'):
+                self.log.info('Getting connection using a JSON key file.')
+                return get_client_from_auth_file(BatchAIManagementClient,
+                                                 key_path)
+            else:
+                raise AirflowException('Unrecognised extension for key file.')
+
+        if os.environ.get('AZURE_AUTH_LOCATION'):
+            key_path = os.environ.get('AZURE_AUTH_LOCATION')
+            if key_path.endswith('.json'):
+                self.log.info('Getting connection using a JSON key file.')
+                return get_client_from_auth_file(BatchAIManagementClient,
+                                                 key_path)
+            else:
+                raise AirflowException('Unrecognised extension for key file.')
+        
+        credentials = ServicePrincipalCredentials(
+            client_id=conn.login,
+            secret=conn.password,
+            tenant=conn.extra_dejson['tenantId']
+        )
+
+        subscription_id = conn.extra_dejson['subscriptionId']
+        return BatchAIManagementClient(credentials, str(subscription_id))
+
+    def create(self, resource_group, workspace_name, cluster_name, location, parameters):
+        print "creating workspace....."
+        self.connection.workspaces._create_initial(resource_group,
+                                                    workspace_name,
+                                                    location)
+        
+        print "creating cluster....."                                            
+        self.connection.clusters._create_initial(resource_group,
+                                          workspace_name,
+                                          cluster_name,
+                                          parameters)
+                                                        
+    def update(self, resource_group, workspace_name, cluster_name):
+        print "updating cluster....."
+        self.connection.clusters.update(resource_group,
+                                          workspace_name,
+                                          cluster_name)
+
+    def get_state_exitcode(self, resource_group, workspace_name, cluster_name):
+        response = self.connection.clusters.get(resource_group,
+                                                        workspace_name,
+                                                        cluster_name,
+                                                        raw=True).response.json()
+        cluster = response['properties']['cluster']
+        instance_view = cluster[0]['properties'].get('instanceView', {})
```

Comment:
can there be multiple clusters? Should we check to ensure there is at least 1? or >1 do we just do the first?